internal/storage/postgres/flowerbed/get_test.go

```diff
@@ -0,0 +1,437 @@
+
+		// then
+		assert.Error(t, err)
+		assert.Nil(t, got)
+	})
+
+	t.Run("should return error when flowerbed with negative id", func(t *testing.T) {
+		// given
+		r := NewFlowerbedRepository(suite.Store, mappers)
+
+		// when
+		got, err := r.GetSensorByFlowerbedID(context.Background(), -1)
+
+		// then
+		assert.Error(t, err)
+		assert.Nil(t, got)
+	})
+
+	t.Run("should return error when flowerbed with negative id", func(t *testing.T) {
+		// given
+		r := NewFlowerbedRepository(suite.Store, mappers)
+
+		// when
+		got, err := r.GetSensorByFlowerbedID(context.Background(), 0)
+
+		// then
+		assert.Error(t, err)
+		assert.Nil(t, got)
+	})
+
+	t.Run("should return error when context is canceled", func(t *testing.T) {
+		// given
+		r := NewFlowerbedRepository(suite.Store, mappers)
+		ctx, cancel := context.WithCancel(context.Background())
+		cancel()
+
+		// when
+		got, err := r.GetSensorByFlowerbedID(ctx, 1)
+
+		// then
+		assert.Error(t, err)
+		assert.Nil(t, got)
+	})
+}
+
+func TestFlowerbedRepository_GetRegionByFlowerbedID(t *testing.T) {
+	suite.ResetDB(t)
+	suite.InsertSeed(t, "internal/storage/postgres/seed/test/flowerbed")
+
+	t.Run("should return linked region to flowerbed", func(t *testing.T) {
+		// given
+		r := NewFlowerbedRepository(suite.Store, mappers)
+
+		// when
+		got, err := r.GetRegionByFlowerbedID(context.Background(), 1)
+
+		expectedRegion := allTestFlowerbeds[0].RegionID
+
+		// then
+		assert.NoError(t, err)
+		assert.NotNil(t, got)
+		assert.NotNil(t, got)
+		assert.Equal(t, expectedRegion, got.ID)
+	})
+
+	t.Run("should return error when flowerbed with non-existing id", func(t *testing.T) {
+		// given
+		r := NewFlowerbedRepository(suite.Store, mappers)
+
+		// when
+		got, err := r.GetRegionByFlowerbedID(context.Background(), 99)
+
+		// then
+		assert.Error(t, err)
+		assert.Nil(t, got)
+	})
+
+	t.Run("should return error when flowerbed with negative id", func(t *testing.T) {
+		// given
+		r := NewFlowerbedRepository(suite.Store, mappers)
+
+		// when
+		got, err := r.GetRegionByFlowerbedID(context.Background(), -1)
+
+		// then
+		assert.Error(t, err)
+		assert.Nil(t, got)
+	})
+
+	t.Run("should return error when flowerbed with negative id", func(t *testing.T) {
+		// given
+		r := NewFlowerbedRepository(suite.Store, mappers)
+
+		// when
+		got, err := r.GetRegionByFlowerbedID(context.Background(), 0)
+
+		// then
+		assert.Error(t, err)
+		assert.Nil(t, got)
+	})
+
+	t.Run("should return error when context is canceled", func(t *testing.T) {
+		// given
+		r := NewFlowerbedRepository(suite.Store, mappers)
+		ctx, cancel := context.WithCancel(context.Background())
+		cancel()
+
+		// when
+		got, err := r.GetRegionByFlowerbedID(ctx, 1)
+
+		// then
+		assert.Error(t, err)
+		assert.Nil(t, got)
+	})
+}
+
+type testFlowerbeds struct {
+	ID          int32
+	Description string
+	RegionID    int32
+	SensorID    int32
+	Images      []*entities.Image
+}
+
+var allTestFlowerbeds = []*testFlowerbeds{
```

Comment:
Maybe seperate the "fake" Flowerbed Data into its own file to have this a bit more organized?